crates/snapbox/src/assert.rs

```diff
@@ -197,11 +197,15 @@ impl Assert {
     ) -> (crate::Result<crate::Data>, crate::Data) {
         let expected = expected.map(|d| d.map_text(crate::utils::normalize_lines));
         // On `expected` being an error, make a best guess
-        if let Some(expected) = expected.as_ref().map(|d| d.as_str()).unwrap_or(Some("")) {
-            actual = actual
-                .try_text()
-                .map_text(crate::utils::normalize_text)
-                .map_text(|t| self.substitutions.normalize(t, expected));
+        if let Ok(ex) = expected.as_ref() {
+            if let Ok(d) = actual.clone().coerce(ex.format()) {
+                actual = d.map_text(crate::utils::normalize_text);
+                if ex.format() == DataFormat::Text {
+                    actual = actual
+                        .clone()
+                        .map_text(|t| self.substitutions.normalize(t, &ex.try_string().unwrap()));
```

Comment:
Why was the `expected.as_str` removed for getting the expected's string?  I would like to see problems that will be solved a different solution not have intermediate changes that muddle the intentions of the current API

Reply:
I switched it because we found that some data formats wouldn't work with `&str` they need to be `String` due to ownership issues. It is one of the reasons `as_str` is supposed to be switched to `try_string` in the long run. I may have mistaken the role of `try_string` here.

Reply:
The usage here ties back into the plans for normalization